File: Aiko Teste/teste/src/main/java/TesteEstagioBackendV2/teste/Model/EquipmentStateHistory.java
package TesteEstagioBackendV2.teste.Model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class EquipmentStateHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "equipment_id")
    private Equipment equipment;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "equipment_state_id")
    private EquipmentState equipmentState;

    @Column
    private Date date;


}

File: Api/Equipaments/Models/Equipment_State_History.cs
using System;

namespace Equipments.Models
{
    public class Equipment_State_History
    {
        public Guid equipment_id { get; set; }
        public Guid equipment_state_id { get; set; }
        public virtual Equipment equipment { get; set; }
        public virtual Equipment_State equipment_state { get; set; }
        public DateTime date { get; set; }
    }
}
